### PR TITLE
doc: manifest: Add a comment explaining how to create a PR

### PR DESCRIPTION
**Pull request by SebastianBoe**
Add a comment explaining how to create a PR.

I keep forgetting the format and would like it documented close to
where I edit the manifest.

Signed-off-by: Sebastian Bøe <sebastian.boe@nordicsemi.no>

### CODE REVIEW

File: west.yml [hunk 1]
```diff
@@ -9,6 +9,10 @@
 # See the west documentation for more information:
 #
 # https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/zephyr/guides/west/index.html
+#
+# When creating a PR, change the SHA to
+# pull/666/head
+# replacing 666 with your PR number
```
[the remark this answers is not in the document]
Comment (SebastianBoe) [reply west.yml:15]: That link does not contain the information that I want.

**Review by mbolivar-nordic — CHANGES_REQUESTED** [review]
We have internal documentation for this that you should bookmark if you are having trouble remembering how to do this. The main manifest file is no place for notes-to-self.

**Comment by SebastianBoe** [comment]
Okay, 2 people who think this is a bad idea. Maybe I'm wrong then.

I had a second idea though.

What if we had a convention of trailing the PR reference in a comment after the SHA.

Then we would be able to easily see which is the latest PR to be merged, which is useful context, and we would also have a quick reference for the format readily available.

      revision: a4edab554f67590a89703067af703c7996604562 # pull/666/head

Is this acceptable @mbolivar-nordic  ?

**Comment by mbolivar-nordic** [comment]
I'm confused because establishing the convention in this proposal seems like more work for the contributor, not less.

We do have a script already (nrf/scripts/get_pulls_in_range.py) that fetches the pull request associated with any commit in any github repository.

In the case of your example, you could invoke:

```
$ ./scripts/get_pulls_in_range.py nrfconnect/sdk-zephyr ../zephyr a4edab554f67590a89703067af703c7996604562~ a4edab554f67590a89703067af703c7996604562
[...]
- #736: [nrf fromlist] twister: short build paths - fix
  https://github.com/nrfconnect/sdk-zephyr/pull/736
```

It would be straightforward to write a function in your .bashrc that gets the pull request associated with a single commit using this helper script if you wanted, and that wouldn't require everyone to maintain comments in the manifest file in a fragile syntax that we couldn't realistically check in CI.

I am very sympathetic to making this workflow easier (for example, I'm positive towards the rumors I keep hearing about an auto-merger), but I don't get why that proposal is helpful, sorry.

**Comment by SebastianBoe** [comment]
"I'm confused because establishing the convention in this proposal seems like more work for the contributor, not less."

I don't think it's more work actually, when the user changes from PR ref to SHA he comments out the ref instead of deleting it, this takes the same amount of time.

The only downside I see is that this will cause review feedback when the user forgets. Which is significant,
so perhaps this is not a good idea.

I am surprised about the pushback here, I'm just asking for a small comment, I wouldn't think it would be so harmful.

Having redundant quick-reference documentation at the top of config files is a common *nix practice.

I tried the PR puller. Very cool :)

But I didn't get the PR format from it:

```
sebo@sebo:~/ncs/nrf$ ./scripts/get_pulls_in_range.py nrfconnect/sdk-nrfxlib ../nrfxlib HEAD~5 HEAD
warning: --sqlite-db was not given; you will need to start over if you are rate-limited or another error occurs
Range HEAD~5..HEAD has 5 commits.
Fetching data for 5 commits from the network...
Missing GitHub credentials:
~/.netrc file not found or has no github.com credentials, and GITHUB_TOKEN is not set in the environment. Please give your GitHub token.

	6195aa85b7 doc: Version update for NCS 1.9.0
	           PR #  642: doc: Version update for NCS 1.9.0
	70dd41fedf crypto: Add PSA supported algs for cc3xx
	           PR #  659: crypto: Add missing PSA driver supported algorithms for Oberon and CC3XX.
	d539dc6864 crypto: Add PSA supported algs for Oberon
	           PR #  659: crypto: Add missing PSA driver supported algorithms for Oberon and CC3XX.
	382af40f50 openthread: add libraries based on commit ce77ab3c
	           PR #  658: openthread: add libraries based on commit ce77ab3c
	414e8d5bc8 nrf_security: Always enable mbedlts RSA builtin
	           PR #  654: nrf_security: Misc legacy fixes for nRF Connect SDK 1.9.0
Done fetching commit data from the network.
- #642: doc: Version update for NCS 1.9.0
  https://github.com/nrfconnect/sdk-nrfxlib/pull/642
- #659: crypto: Add missing PSA driver supported algorithms for Oberon and CC3XX.
  https://github.com/nrfconnect/sdk-nrfxlib/pull/659
- #658: openthread: add libraries based on commit ce77ab3c
  https://github.com/nrfconnect/sdk-nrfxlib/pull/658
- #654: nrf_security: Misc legacy fixes for nRF Connect SDK 1.9.0
  https://github.com/nrfconnect/sdk-nrfxlib/pull/654

```

The PR format is not in the output, and I couldn't find any branches (using magit) in the nrfxlib repo either.

**Comment by SebastianBoe** [comment]
> It would be straightforward to write a function in your .bashrc that gets the pull request associated with a single commit using this helper script if you wanted, and that wouldn't require everyone to maintain comments in the manifest file in a fragile syntax that we couldn't realistically check in CI.


Ah, I see, okay, I now have this script

sebo@sebo:~/ncs/nrf$ ~/pr
pull/6/head
sebo@sebo:~/ncs/nrf$ cat ~/pr
#!/bin/bash
echo "pull/6/head"

I was hoping to solve the problem for more people, but it seems this is not possible.

**Comment by SebastianBoe** [comment]
Ah, I misunderstood you, sorry, you proposed using get_pulls_in_range.py to find the link between commits and pr refs.

But I thought you were proposing to use it to find the pr ref format.

So I created my own script for that, since this PR was denied.

**Pull request closed without merging**